### ISSUE TITLE
Any plans for a "torch.minmax" (min-max normalization) function?

### ISSUE THREAD

**Issue by rafa-br34** [pytorch/pytorch#128785]
### 🚀 The feature, motivation and pitch

While min-max normalization is an easy function to implement, I think having it by default on Pytorch could be a great addition.
(Forgive me if this was already asked)

**Comment by vadimkantorov** [comment]
Related:
- https://github.com/pytorch/pytorch/issues/107896

**Comment by albanD** [comment]
Hey!
Doesn't https://pytorch.org/docs/stable/generated/torch.aminmax.html#torch-aminmax do what you want?

**Comment by rafa-br34** [comment]
> Hey! Doesn't https://pytorch.org/docs/stable/generated/torch.aminmax.html#torch-aminmax do what you want?

I didn't test but from reading the documentation it seems to return a tuple with the lowest and the highest value found in the tensor. I want min-max normalization, take [this StackOverflow post](https://stackoverflow.com/a/68791573/14877269) as an example.

**Comment by AndhikaWB** [comment]
I created a `MinMax` class based on this PyTorch forum [discussion](https://discuss.pytorch.org/t/using-scikit-learns-scalers-for-torchvision/53455) and Sklearn implementation [[1](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MinMaxScaler.html), [2](https://github.com/scikit-learn/scikit-learn/blob/ddc170d41/sklearn/preprocessing/_data.py#L291)]:

``` python
class MinMax(torch.nn.Module):
    def __init__(self, fit_tensor: torch.Tensor, feature_range = (0, 1), inplace = False):
        super().__init__()
        self.feature_range = feature_range
        self.inplace = inplace
        self.fit_(fit_tensor)

    def fit_(self, tensor: torch.Tensor):
        # Get min and max values for each variable
        data_min = tensor.min(dim = 0, keepdim = True)[0]
        data_max = tensor.max(dim = 0, keepdim = True)[0]

        data_range = data_max - data_min
        # Prevent division by zero in line below
        data_range[data_range == 0.0] = 1.0

        # See the Sklearn implementation for reference
        self.scale_ = (self.feature_range[1] - self.feature_range[0]) / data_range
        self.min_ = self.feature_range[0] - data_min * self.scale_

        self.data_min_ = data_min
        self.data_max_ = data_max
        self.data_range_ = data_range

        return self

    def forward(self, tensor: torch.Tensor):
        if not self.inplace:
            tensor = tensor.clone()

        tensor.mul_(self.scale_)
        tensor.add_(self.min_)

        return tensor
    
    def backward(self, tensor: torch.Tensor):
        if not self.inplace:
            tensor = tensor.clone()

        tensor.sub_(self.min_)
        tensor.div_(self.scale_)

        return tensor
```

Seems to work well after a simple testing:

``` python
arr = torch.Tensor([
    [7,2,3],
    [4,5,9],
    [1,8,6]
])

temp = MinMax(fit_tensor = arr)
fwd_arr = temp(arr)
bwd_arr = temp.backward(fwd_arr)

print(arr)
print(fwd_arr)
print(bwd_arr)
```

```
tensor([[7., 2., 3.],
        [4., 5., 9.],
        [1., 8., 6.]])
tensor([[1.0000, 0.0000, 0.0000],
        [0.5000, 0.5000, 1.0000],
        [0.0000, 1.0000, 0.5000]])
tensor([[7.0000, 2.0000, 3.0000],
        [4.0000, 5.0000, 9.0000],
        [1.0000, 8.0000, 6.0000]])
```

Not sure if this follows PyTorch style (I'm new to PyTorch), but just want to share in case someone need a quick solution
